package/test/index.test.ts

```diff
@@ -170,6 +171,47 @@ describe('requestHandler', () => {
     });
   });
 
+  /**
+   * Test excessive pagination returns 400
+   */
+
+  it('should return 400', async () => {
+    // Prerequisites
+    await initTestDatabase(initTestDatabaseParams);
+    await createColumnAndIndex({ tableName });
+
+    // Spy on console.log so errors don't polute the test output
+    const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
+
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn().mockReturnThis(),
+    };
+
+    const req = {
+      body: { params: { query: 'good', page: 200 }, indexName: tableName },
+    };
+    await searchHandler(req, res);
+
+    // Test status and json are called
+    expect(consoleSpy).toHaveBeenCalled();
+    expect(res.status).toHaveBeenCalledWith(400);
+    expect(res.json).toHaveBeenCalledWith({
+      error: 'Request contained invalid payload',
```

Comment:
You're right, and this has been refactored to log the errors in https://github.com/dekimir/postgres-searchbox/pull/12 , but return a vague error to the end user to prevent data leak.

Reply:
I don't see how this can be a data leak. The validation logic is public.

Reply:
Oh that's my bad, so any / all zod validation errors should be returnded to the client. It would certainly be helpful to the end developer.
Do you think zod is actually the **only** error message that should be passed to the client. I'm thinking that a database or other error should be simplified to prevent a leak of stack info.

Reply:
> Do you think zod is actually the **only** error message that should be passed to the client.

Yes.

> I'm thinking that a database or other error should be simplified to prevent a leak of stack info.

Exactly.